lib/cartopy/mpl/geoaxes.py

```diff
@@ -784,6 +796,187 @@ def stock_img(self, name='ne_shaded'):
+                ret_extent = [lon_pts[lon_in_range1][0] - d_lon / 2.0,
+                              lon_pts[lon_in_range2][-1] + d_lon / 2.0 + 360,
+                              lat_pts[lat_in_range][-1] - d_lat / 2.0,
+                              lat_pts[lat_in_range][0] + d_lat / 2.0]
+            else:
+                # not crossing the dateline, so just find the region:
+                lon_in_range = np.logical_and(lon_pts >= extent[0],
+                                              lon_pts <= extent[1])
+                img_subset = img[lat_in_range, :, :][:, lon_in_range, :]
+                # now define the extent for output that matches those points:
+                ret_extent = [lon_pts[lon_in_range][0] - d_lon / 2.0,
+                              lon_pts[lon_in_range][-1] + d_lon / 2.0,
+                              lat_pts[lat_in_range][-1] - d_lat / 2.0,
+                              lat_pts[lat_in_range][0] + d_lat / 2.0]
+
+            return self.imshow(img_subset, origin='upper',
+                               transform=source_proj,
+                               extent=ret_extent)
+
+    def read_user_backgound_images(self, verify=True):
+        """
+        Reads the metadata in the specified CARTOPY_USER_BACKGROUNDS
+        environment variable to populate the dictionaries for background_img.
+
+        If CARTOPY_USER_BACKGROUNDS is not set then by default the image in
+        lib/cartopy/data/raster/natural_earth/ will be made available.
+
+        The metadata should be a standard JSON file which specifies a two
+        level dictionary. The first level is the image type.
+        For each image type there needs to be the fields:
+        __comment__, __source__ and __projection__
+        and then an element giving the filename for each resolution.
+
+        An example JSON file can be found at:
+        lib/cartopy/data/raster/natural_earth/images.json
+
+        """
+        import os
+        import json
+
+        bgdir = os.getenv('CARTOPY_USER_BACKGROUNDS')
+        if bgdir is None:
+            bgdir = os.path.join(config["repo_data_dir"],
+                                 'raster', 'natural_earth')
+        json_file = os.path.join(bgdir, 'images.json')
+
+        with open(json_file, 'r') as js_obj:
+            dict_in = json.load(js_obj)
+        for img_type in dict_in:
+            _USER_BG_IMGS[img_type] = dict_in[img_type]
+
+        if verify:
+            required_info = ['__comment__', '__source__', '__projection__']
+            for img_type in _USER_BG_IMGS:
+                if img_type == '__comment__':
+                    # the top level comment doesn't need verifying:
+                    pass
+                else:
+                    # check that this image type has the required info:
+                    for required in required_info:
+                        if required not in _USER_BG_IMGS[img_type]:
+                            err_str = 'User background metadata file'
```

Comment:
I don't like it either but the pep8 compliance demands short lines, so it's a choice between useful error messages and += or useless error messages (or ignoring pep8 compliance).